jnigen/lib/src/bindings/c_bindings.dart

```diff
@@ -72,50 +72,51 @@ class CBindingGenerator {
     final s = StringBuffer();
     final name = m.finalName;
     final functionName = getMemberNameInC(c, name);
+    final classRef = '${classVarPrefix}_$classNameInC';
     final methodID = '${methodVarPrefix}_$functionName';
-    s.write('jmethodID $methodID = NULL;\n');
-
     final cMethodName = getMemberNameInC(c, name);
-    final cParams = _formalArgs(m);
-    s.write('FFI_PLUGIN_EXPORT\n');
-    s.write('$jniResultType $cMethodName($cParams) {\n');
-
-    final classVar = '${classVarPrefix}_$classNameInC';
+    final cMethodParams = _formalArgs(m);
     final jniSignature = getJniSignatureForMethod(m);
+    final ifStaticMethodID = isStatic ? 'static_' : '';
 
-    s.write(_loadEnvCall);
-    s.write(_loadClassCall(classVar, getInternalName(c.binaryName)));
-
-    final ifStatic = isStatic ? 'static_' : '';
-    s.write('${indent}load_${ifStatic}method($classVar, '
-        '&$methodID, "${m.name}", "$jniSignature");\n');
-    s.write('${indent}if ($methodID == NULL) return $ifError;\n');
-
-    var returnTypeName = m.returnType.name;
+    var javaReturnType = m.returnType.name;
     if (isACtor) {
-      returnTypeName = c.binaryName;
-    }
-
-    s.write(indent);
-    if (returnTypeName != 'void') {
-      s.write('${getCType(returnTypeName)} _result = ');
+      javaReturnType = c.binaryName;
     }
     final callType = _typeNameAtCallSite(m.returnType);
-    final callArgs = _callArgs(m, classVar, methodID);
-    if (isACtor) {
-      s.write('(*jniEnv)->NewObject($callArgs);\n');
-    } else {
-      final ifStatic = isStatic ? 'Static' : '';
-      s.write('(*jniEnv)->Call$ifStatic${callType}Method($callArgs);\n');
+    final callArgs = _callArgs(m, classRef, methodID);
+
+    var ifAssignResult = '';
+    if (javaReturnType != 'void') {
+      ifAssignResult = '${getCType(javaReturnType)} _result = ';
     }
-    s.write(_result(m));
-    s.write('}\n');
+
+    final ifStaticCall = isStatic ? 'Static' : '';
+    final envMethod =
+        isACtor ? 'NewObject' : 'Call$ifStaticCall${callType}Method';
+    final returnResultIfAny = _result(m);
+    s.write('jmethodID $methodID = NULL;\n'
```

Comment:
This would be even clearer as a multi-line string:

```dart
s.write('''
jmethodID $methodID = NULL;
FFI_PLUGIN_EXPORT
$jniResultType $cMethodName($cMethodParams) {

...

  ''');
```

Reply:
One thing I find annoying with multiline strings is we have to indent back. Is there  a rule of thumb when to use multiline strings and when to use normal strings?

Reply:
The rule of thumb is probably just "which approach makes my code clearer to read?"